pkg/virt-operator/kubevirt.go

```diff
@@ -398,16 +401,16 @@ func (c *KubeVirtController) execute(key string) error {
 	kvCopy := kv.DeepCopy()
 
 	if kv.DeletionTimestamp != nil {
-		syncError = c.syncDeletion(kvCopy)
+		syncError = c.syncDeletion(kv)
```

Comment:
We have to pass the kvCopy here. The the original kv object we retrieve from the cache should never be modified. That's one of the primary reasons we're making the copy.

If we modify kv here, we'd actually be modifying an object owned and updated by the informer.

Reply:
I don't understand. Were is the difference between making changes to the original resource or to the copy, when doing the update later on...?
The reason I changed this is that I want to check the phase and condition in the unit test. When doing changes to the copy, that doesn't work.

Reply:
> modifying an object owned and updated by the informer.

ah, got it...

Reply:
> ah, got it...

yeah, we don't do enough to protect ourselves here. The only reason I know about this is because I made this mistake and couldn't figure out why weird things were occurring later on. It's definitely not obvious.